### ISSUE TITLE
Revamp dokku run command

### ISSUE THREAD

**Issue by tylercal** [dokku/dokku#4699]
# High Level Plan

I'm going to split this up into a `dokku run` and `dokku run:detached`. There are definitely some weird things about using it.

- Both commands will remove the containers by default. No option will be available to persist the container (scale a`bash` or `console` process defined in your Procfile if you want something persistent) as the default behavior is now
- `run:detached` will additionally add the `-d` flag (run the container in detached mode)
- Remove the `--detach` flag
- Remove the `--rm` and `--rm-container` flags
- Only `run` will use a tty when available
- A new `--no-tty` flag will be added to `run` to optionally disable a tty (no effect if there wasn't a tty in the first place).

# Original Post

## Description of problem
About every 5 minutes, I get a flood of messages in my logs along the lines of `Attempting to retire epta image 6f93d2aa2a5....4c4ae5d8e74c16b3` `Image 6f93d2aa2a5....4c4ae5d8e74c16b3 still running` (container image is shortened)

Upon checking with `docker ps -a` I can see that all of these are from old `dokku run` commands (usually doing something like running a rails console). When I `docker inspect` a couple, they all have "Running" and "Restarting" set to false, so I'd think the images _should_ be able to be retired.

My best guess is that these are commands that were running in a session where my ssh connection to the server timed out and disconnected ending the process.

### `dokku report APP_NAME` output
This does affect multiple applications, but here is the report for the app with the most zombie processes
```
~> dokku report naked-game
-----> uname: Linux DokkuVM 4.15.0-151-generic #157-Ubuntu SMP Fri Jul 9 23:07:57 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
-----> memory: 
                     total        used        free      shared  buff/cache   available
       Mem:           7977        1329        3487          77        3160        6309
       Swap:             0           0           0
-----> docker version: 
       Client: Docker Engine - Community
        Version:           19.03.13
        API version:       1.40
        Go version:        go1.13.15
        Git commit:        4484c46d9d
        Built:             Wed Sep 16 17:02:59 2020
        OS/Arch:           linux/amd64
        Experimental:      false
       
       Server: Docker Engine - Community
        Engine:
         Version:          19.03.13
         API version:      1.40 (minimum version 1.12)
         Go version:       go1.13.15
         Git commit:       4484c46d9d
         Built:            Wed Sep 16 17:01:30 2020
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.3.7
         GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
        runc:
         Version:          1.0.0-rc10
         GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
        docker-init:
         Version:          0.18.0
         GitCommit:        fec3683
-----> docker daemon info: 
       Client:
        Debug Mode: true
       
       Server:
WARNING: No swap limit support
        Containers: 30
         Running: 17
         Paused: 0
         Stopped: 13
        Images: 77
        Server Version: 19.03.13
        Storage Driver: overlay2
         Backing Filesystem: extfs
         Supports d_type: true
         Native Overlay Diff: true
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: 8fba4e9a7d01810a393d5d25a3621dc101981175
        runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
        init version: fec3683
        Security Options:
         apparmor
         seccomp
          Profile: default
        Kernel Version: 4.15.0-151-generic
        Operating System: Ubuntu 18.04.5 LTS
        OSType: linux
        Architecture: x86_64
        CPUs: 2
        Total Memory: 7.79GiB
        Name: DokkuVM
        ID: CA3Z:GOHT:SO22:RWHD:BQCZ:QF4Y:ZIUH:POZD:HOZE:D7LY:HD6Y:YQCR
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
        Registry: https://index.docker.io/v1/
        Labels:
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false
       
-----> git version: git version 2.17.1
-----> sigil version: 0.6.0
-----> herokuish version: 
       herokuish: 0.5.29
       buildpacks:
         heroku-buildpack-multi     v1.0.0
         heroku-buildpack-ruby      v228
         heroku-buildpack-nodejs    v185
         heroku-buildpack-clojure   v87
         heroku-buildpack-python    v197
         heroku-buildpack-java      v69
         heroku-buildpack-gradle    v35
         heroku-buildpack-scala     v90
         heroku-buildpack-play      v26
         heroku-buildpack-php       v195
         heroku-buildpack-go        v153
         buildpack-nginx            v14
         buildpack-null             v3
-----> dokku version: dokku version 0.24.10
-----> plugn version: plugn: 0.6.1
-----> dokku plugins: 
         00_dokku-standard    0.24.10 enabled    dokku core standard plugin
         20_events            0.24.10 enabled    dokku core events logging plugin
         app-json             0.24.10 enabled    dokku core app-json plugin
         apps                 0.24.10 enabled    dokku core apps plugin
         builder              0.24.10 enabled    dokku core builder plugin
         builder-dockerfile   0.24.10 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.24.10 enabled    dokku core builder-herokuish plugin
         builder-pack         0.24.10 enabled    dokku core builder-pack plugin
         buildpacks           0.24.10 enabled    dokku core buildpacks plugin
         certs                0.24.10 enabled    dokku core certificate management plugin
         checks               0.24.10 enabled    dokku core checks plugin
         common               0.24.10 enabled    dokku core common plugin
         config               0.24.10 enabled    dokku core config plugin
         cron                 0.24.10 enabled    dokku core cron plugin
         docker-options       0.24.10 enabled    dokku core docker-options plugin
         domains              0.24.10 enabled    dokku core domains plugin
         enter                0.24.10 enabled    dokku core enter plugin
         git                  0.24.10 enabled    dokku core git plugin
         hostname             0.2.0 enabled    Sets the docker hostname option for dokku
         letsencrypt          0.11.9 enabled    Automated installation of let's encrypt TLS certificates
         logs                 0.24.10 enabled    dokku core logs plugin
         logspout             0.4.0 enabled    sends dokku app stdout to a logging service
         network              0.24.10 enabled    dokku core network plugin
         nginx-vhosts         0.24.10 enabled    dokku core nginx-vhosts plugin
         plugin               0.24.10 enabled    dokku core plugin plugin
         postgres             1.0.0 enabled    dokku postgres service plugin
         proxy                0.24.10 enabled    dokku core proxy plugin
         ps                   0.24.10 enabled    dokku core ps plugin
         redirect             0.7.1 enabled    Plugin for managing application redirects
         redis                1.12.5 enabled    dokku redis service plugin
         repo                 0.24.10 enabled    dokku core repo plugin
         resource             0.24.10 enabled    dokku core resource plugin
         scheduler-docker-local 0.24.10 enabled    dokku core scheduler-docker-local plugin
         shell                0.24.10 enabled    dokku core shell plugin
         ssh-keys             0.24.10 enabled    dokku core ssh-keys plugin
         storage              0.24.10 enabled    dokku core storage plugin
         tags                 0.24.10 enabled    dokku core tags plugin
         tar                  0.24.10 enabled    dokku core tar plugin
         trace                0.24.10 enabled    dokku core trace plugin
=====> naked-game app information
       App deploy source:             
       App dir:                       /home/dokku/naked-game
       App locked:                    false
=====> naked-game builder information
       Builder computed selected:     
       Builder global selected:       
       Builder selected:              
=====> naked-game buildpacks information
       Buildpacks computed stack:     gliderlabs/herokuish:latest
       Buildpacks global stack:       
       Buildpacks list:               
       Buildpacks stack:              
=====> naked-game ssl information
       Ssl dir:                       /home/dokku/naked-game/tls
       Ssl enabled:                   true                     
       Ssl hostnames:                 naked.roomj.com playthenakedgame.com
       Ssl expires at:                Sep 22 20:03:22 2021 GMT 
       Ssl issuer:                    C = US, O = Let's Encrypt, CN = R3
       Ssl starts at:                 Jun 24 20:03:23 2021 GMT 
       Ssl subject:                   subject=CN = naked.roomj.com
       Ssl verified:                  verified by a certificate authority
=====> naked-game checks information
       Checks disabled list:          none                     
       Checks skipped list:           none                     
=====> naked-game cron information
       Cron task count:               1
=====> naked-game docker options information
       Docker options build:          --link dokku.postgres.naked-game:dokku-postgres-naked-game --link dokku.redis.naked-game:dokku-redis-naked-game 
       Docker options deploy:         --link dokku.postgres.naked-game:dokku-postgres-naked-game --link dokku.redis.naked-game:dokku-redis-naked-game --restart=on-failure:10 
       Docker options run:            --link dokku.postgres.naked-game:dokku-postgres-naked-game --link dokku.redis.naked-game:dokku-redis-naked-game 
=====> naked-game domains information
       Domains app enabled:           true                     
       Domains app vhosts:            naked.roomj.com playthenakedgame.com
       Domains global enabled:        true                     
       Domains global vhosts:         snapbiller.westus2.cloudapp.azure.com dokku.roomj.com
=====> naked-game git information
       Git deploy branch:             master                   
       Git global deploy branch:      master                   
       Git keep git dir:              false                    
       Git rev env var:               GIT_REV                  
       Git sha:                       0c00f8c                  
       Git last updated at:           1626985721               
=====> naked-game logs information
       Logs computed max size:        10m
       Logs global max size:          10m
       Logs global vector sink:       
       Logs max size:                 
       Logs vector sink:              
=====> naked-game network information
       Network attach post create:    
       Network attach post deploy:    
       Network bind all interfaces:   false
       Network web listeners:         172.17.0.9:5000
=====> naked-game nginx information
       Nginx access log format:                                
       Nginx access log path:         /var/log/nginx/naked-game-access.log
       Nginx bind address ipv4:                                
       Nginx bind address ipv6:       ::                       
       Nginx client max body size:                             
       Nginx disable custom config:   false                    
       Nginx error log path:          /var/log/nginx/naked-game-error.log
       Nginx global hsts:             true                     
       Nginx computed hsts:           true                     
       Nginx hsts:                                             
       Nginx hsts include subdomains: true                     
       Nginx hsts max age:            15724800                 
       Nginx hsts preload:            false                    
       Nginx proxy buffer size:       4096                     
       Nginx proxy buffering:         on                       
       Nginx proxy buffers:           8 4096                   
       Nginx proxy busy buffers size: 8192                     
       Nginx proxy read timeout:      60s                      
       Nginx last visited at:         1627209802               
       Nginx x forwarded for value:   $remote_addr             
       Nginx x forwarded port value:  $server_port             
       Nginx x forwarded proto value: $scheme                  
       Nginx x forwarded ssl:                                  
=====> naked-game proxy information
       Proxy enabled:                 true
       Proxy port map:                http:80:5000 https:443:5000
       Proxy type:                    nginx
=====> naked-game ps information
       Deployed:                      true
       Processes:                     1
       Ps can scale:                  true
       Ps restart policy:             on-failure:10
       Restore:                       true
       Running:                       true
       Status web 1:                  running (CID: d2533c54d6f)
=====> naked-game resource information
=====> naked-game scheduler-docker-local information
       Scheduler docker local disable chown:                          
=====> naked-game storage information
       Storage build mounts:                                   
       Storage deploy mounts:                                  
       Storage run mounts:  
```

### How (deb/make/rpm) and where (AWS, VirtualBox, physical, etc.) was Dokku installed?
Installed on Azure VM, Ubuntu via apt

### Additional information
`sudo docker ps -a` (excerpt)
```
CONTAINER ID        IMAGE                      COMMAND                  CREATED             STATUS                          PORTS               NAMES
52705d88a16b        16a0cee7d341               "/exec rails console"    5 months ago        Exited (1) 53 minutes ago                           naked-game.run.7777
3c02d8b598c6        49401cd9675a               "/exec rails console"    5 months ago        Exited (1) 53 minutes ago                           naked-game.run.11166
a697322e0162        1cca732a6424               "/exec rails console"    5 months ago        Exited (1) 53 minutes ago                           naked-game.run.22740
7cb22bc4e2b2        4e5ca3946052               "rails console"          7 months ago        Created                                             naked-game.run.20317
6f1a27f6b389        4e5ca3946052               "rails console"          7 months ago        Created                                             naked-game.run.953
```

`docker inspect 52705d88a16b`
```
[
    {
        "Id": "52705d88a16b6940a1ecf27f933b1abe60d961922167f7d9131ca9b16f408c4a",
        "Created": "2021-02-12T20:47:13.916701269Z",
        "Path": "/exec",
        "Args": [
            "rails",
            "console"
        ],
        "State": {
            "Status": "exited",
            "Running": false,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 1,
            "Error": "",
            "StartedAt": "2021-02-12T20:47:14.697624031Z",
            "FinishedAt": "2021-07-25T15:18:17.424128676Z"
        },
        "Image": "sha256:16a0cee7d341495204564082c6dafa1daeb62c1baa5c170537415f3186d370cc",
        "ResolvConfPath": "/var/lib/docker/containers/52705d88a16b6940a1ecf27f933b1abe60d961922167f7d9131ca9b16f408c4a/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/52705d88a16b6940a1ecf27f933b1abe60d961922167f7d9131ca9b16f408c4a/hostname",
        "HostsPath": "/var/lib/docker/containers/52705d88a16b6940a1ecf27f933b1abe60d961922167f7d9131ca9b16f408c4a/hosts",
        "LogPath": "/var/lib/docker/containers/52705d88a16b6940a1ecf27f933b1abe60d961922167f7d9131ca9b16f408c4a/52705d88a16b6940a1ecf27f933b1abe60d961922167f7d9131ca9b16f408c4a-json.log",
        "Name": "/naked-game.run.7777",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "docker-default",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": null,
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {},
            "RestartPolicy": {
                "Name": "no",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "Capabilities": null,
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": [
                "/dokku.postgres.naked-game:/naked-game.run.7777/dokku-postgres-naked-game",
                "/dokku.redis.naked-game:/naked-game.run.7777/dokku-redis-naked-game"
            ],
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/6ce3681e7fea53255ff6109e5a1c30cbf3ff150d2aa30b53452afa75ee8a2396-init/diff:/var/lib/docker/overlay2/cd5f28e0c86edb7c748819f1f52c7e6524e9dbcb33ab29f12aeaf94818ce87a7/diff:/var/lib/docker/overlay2/a093fd466c261ea0ca3957fc0e96b26667e8f02caac1c219b4e3043ca362b15a/diff:/var/lib/docker/overlay2/5e56689fa6b7d8ba02e987809d58b97299707f84318ebee4fbfd4b5631c93975/diff:/var/lib/docker/overlay2/b75652cac8b9ef3faf7ddd803be75c70677b48c281bc6b656cf76fe41eb47ed8/diff:/var/lib/docker/overlay2/c24ad72ce10e1ee365980ed521bea5656470c00be8f77efb6d62678ff66f7c73/diff:/var/lib/docker/overlay2/796403b1892cc51543e8490892f5f4a5e1b8a7cfb4356db8e832e835b875d307/diff:/var/lib/docker/overlay2/e116cc82a5ad33b5235330d0065f8c7bbad82d1102f3ce5370cff49f879d1576/diff:/var/lib/docker/overlay2/20c03751f33be2c0629a2bd5b6627ef88a57a183a94c93c9eab7e501058e4e6f/diff:/var/lib/docker/overlay2/0a56b50305e8dc257937758a1dfef668ac34da8c91769526cf904b6343c83a9b/diff:/var/lib/docker/overlay2/516ac42b90754241aa467042dd0077a9cf71ecc1a83d66c1cb3ca9ea40d9a0ce/diff:/var/lib/docker/overlay2/94d457a5a2b44d7cc34db16ba9329f0d819f4e4cce82eedcfb87fe269cfc4248/diff:/var/lib/docker/overlay2/4e4167e38011223f5ba16b2c5650650026f10d407b86ad3d3418548df1a568ac/diff:/var/lib/docker/overlay2/222dd6f2b53ed6c526fc0946606f2ee8545bb61a72bd5c68eaa66677202c1dcc/diff:/var/lib/docker/overlay2/532b175792c901f93ad5fd14ebb05f7a210ac7799229dcc715335749ac52b6f6/diff:/var/lib/docker/overlay2/bdb4695e5a9d73ad6523ca202528cecd9781ed4c435c44df1d0807611a625f50/diff:/var/lib/docker/overlay2/8d14de6bebcacdda1e35035243c571c8808a5daabf1830b546cd75bbb96a182b/diff:/var/lib/docker/overlay2/a4f766d4783e4a9ef2790036d38c7907fcb78c12de0e8c04791257a851b16eec/diff:/var/lib/docker/overlay2/910a19b43f520d23e91110d5ce1b4b5a130b28ccf86cb155dcf51696f16bca51/diff",
                "MergedDir": "/var/lib/docker/overlay2/6ce3681e7fea53255ff6109e5a1c30cbf3ff150d2aa30b53452afa75ee8a2396/merged",
                "UpperDir": "/var/lib/docker/overlay2/6ce3681e7fea53255ff6109e5a1c30cbf3ff150d2aa30b53452afa75ee8a2396/diff",
                "WorkDir": "/var/lib/docker/overlay2/6ce3681e7fea53255ff6109e5a1c30cbf3ff150d2aa30b53452afa75ee8a2396/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [],
        "Config": {
            "Hostname": "DokkuVM",
            "Domainname": "",
            "User": "",
            "AttachStdin": true,
            "AttachStdout": true,
            "AttachStderr": true,
            "Tty": true,
            "OpenStdin": true,
            "StdinOnce": true,
            "Env": [
                "USER=herokuishuser",
                "DYNO=run.7777",
                "CACHE_PATH=/cache",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "STACK=heroku-18",
                "DEBIAN_FRONTEND=noninteractive"
            ],
            "Cmd": [
                "/exec",
                "rails",
                "console"
            ],
            "Image": "dokku/naked-game:latest",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "com.dokku.app-name": "naked-game",
                "com.dokku.container-type": "run",
                "com.dokku.image-stage": "release",
                "com.gliderlabs.herokuish/stack": "heroku-18",
                "dokku": "",
                "org.label-schema.schema-version": "1.0",
                "org.label-schema.vendor": "dokku"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "3cdccd07b8875503cdad94cbd74e4a8ae441c6423d56bbfa4feaa06723043153",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/3cdccd07b887",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "bf239ac1700c87e3187e243e34c901de5b800cfad3ce446eee5043965301bbad",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
    }
]

```
`docker inspect  6f1a27f6b389`
```
[
    {
        "Id": "6f1a27f6b38949040f34c2e5df279d5497f99acb683e0a08f8c5c350e937c27d",
        "Created": "2020-12-13T03:17:51.233635321Z",
        "Path": "rails",
        "Args": [
            "console"
        ],
        "State": {
            "Status": "created",
            "Running": false,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 127,
            "Error": "OCI runtime create failed: container_linux.go:349: starting container process caused \"exec: \\\"rails\\\": executable file not found in $PATH\": unknown",
            "StartedAt": "0001-01-01T00:00:00Z",
            "FinishedAt": "0001-01-01T00:00:00Z"
        },
        "Image": "sha256:4e5ca3946052c379de6e333de52b0701f0e96ccabb98426167a55fd54688b40b",
        "ResolvConfPath": "/var/lib/docker/containers/6f1a27f6b38949040f34c2e5df279d5497f99acb683e0a08f8c5c350e937c27d/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/6f1a27f6b38949040f34c2e5df279d5497f99acb683e0a08f8c5c350e937c27d/hostname",
        "HostsPath": "/var/lib/docker/containers/6f1a27f6b38949040f34c2e5df279d5497f99acb683e0a08f8c5c350e937c27d/hosts",
        "LogPath": "/var/lib/docker/containers/6f1a27f6b38949040f34c2e5df279d5497f99acb683e0a08f8c5c350e937c27d/6f1a27f6b38949040f34c2e5df279d5497f99acb683e0a08f8c5c350e937c27d-json.log",
        "Name": "/naked-game.run.953",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "docker-default",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": null,
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {},
            "RestartPolicy": {
                "Name": "no",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "Capabilities": null,
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": [
                "/dokku.postgres.naked-game:/naked-game.run.953/dokku-postgres-naked-game",
                "/dokku.redis.naked-game:/naked-game.run.953/dokku-redis-naked-game"
            ],
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": [],
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/6cf92103ae0305e709a8a6228cee3ae268abd6f929121334d2bffadd988e7d09-init/diff:/var/lib/docker/overlay2/1cd5fbe3375c79a0cc28067ae1a9085e8fdce38b5fd8468cbfaba59eccf26e40/diff:/var/lib/docker/overlay2/3dca84813cec363a3c8551e397859c46604ac8a0097f6367a1e9d483569656ab/diff:/var/lib/docker/overlay2/cb69269524a3019745d90145bff5851ada227d16ad6f4511e368622fa5db93c3/diff:/var/lib/docker/overlay2/a2ad0f98f13c69c60fddd2c711e21c0f62748aec52b7580dbbbc32385a9acaa9/diff:/var/lib/docker/overlay2/19a85f056a05afa02f1060797116ef432bd3210bb72f26dd60bad40e16e8f42e/diff:/var/lib/docker/overlay2/6e001acfdb035772285d9c76229cf3d0f4b8b63dc713d4cc65f543262682baeb/diff:/var/lib/docker/overlay2/421f499f3efed7fd2eaa01b19fb92a6ae256fbba5134a8adf1fcddf900a60a00/diff:/var/lib/docker/overlay2/d07baf78dc6e9fbb9b671f85e104e99008909fc32f4f2e5da2b24651f0405355/diff:/var/lib/docker/overlay2/a5f74e45084889db250224f25fa05bf2c477298d4c6e142fea5ecc5a9075e5d9/diff:/var/lib/docker/overlay2/31651bace2c7cb76d15872a99fe4644d241110ea1985ad4c1a432fe726098634/diff:/var/lib/docker/overlay2/b6ac37e9c6fa77b06c41fd09231a2c18aa32d3e7661a15ff76560b8071eb8f58/diff:/var/lib/docker/overlay2/b839b04295a6f77ae28f80c04f60a272af08d84efea583f4e33ca7e94536716e/diff:/var/lib/docker/overlay2/8d14de6bebcacdda1e35035243c571c8808a5daabf1830b546cd75bbb96a182b/diff:/var/lib/docker/overlay2/a4f766d4783e4a9ef2790036d38c7907fcb78c12de0e8c04791257a851b16eec/diff:/var/lib/docker/overlay2/910a19b43f520d23e91110d5ce1b4b5a130b28ccf86cb155dcf51696f16bca51/diff",
                "MergedDir": "/var/lib/docker/overlay2/6cf92103ae0305e709a8a6228cee3ae268abd6f929121334d2bffadd988e7d09/merged",
                "UpperDir": "/var/lib/docker/overlay2/6cf92103ae0305e709a8a6228cee3ae268abd6f929121334d2bffadd988e7d09/diff",
                "WorkDir": "/var/lib/docker/overlay2/6cf92103ae0305e709a8a6228cee3ae268abd6f929121334d2bffadd988e7d09/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [],
        "Config": {
            "Hostname": "DokkuVM",
            "Domainname": "",
            "User": "",
            "AttachStdin": true,
            "AttachStdout": true,
            "AttachStderr": true,
            "Tty": true,
            "OpenStdin": true,
            "StdinOnce": true,
            "Env": [
                "DATABASE_URL=postgres://postgres:30e4f2e7ac94716999fae5f5df38b46c@dokku-postgres-naked-game:5432/naked_game",
                "DOKKU_APP_TYPE=herokuish",
                "DOKKU_PROXY_PORT=80",
                "DOKKU_PROXY_PORT_MAP=http:80:5000 https:443:5000",
                "DOKKU_PROXY_SSL_PORT=443",
                "GIT_REV=78e5c46253bfbd694942a5883abebda01c5410d9",
                "USER=herokuishuser",
                "CURL_CONNECT_TIMEOUT=90",
                "CURL_TIMEOUT=60",
                "DOKKU_APP_RESTORE=1",
                "DOKKU_LETSENCRYPT_EMAIL=tylercal@gmail.com",
                "REDIS_URL=redis://naked-game:a8b9daea0421e9c96df500ef9d6153fd5ef2edb705743d2b3d9b27db8cee1d29@dokku-redis-naked-game:6379",
                "DYNO=run.953",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "DEBIAN_FRONTEND=noninteractive"
            ],
            "Cmd": [
                "rails",
                "console"
            ],
            "Image": "dokku/naked-game:latest",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "com.dokku.app-name": "naked-game",
                "com.dokku.container-type": "run",
                "dokku": "",
                "org.label-schema.schema-version": "1.0",
                "org.label-schema.vendor": "dokku"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "598212d933ba40a9c1e341739406c291f20137653b8e3b437d0348b13554bc7a",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/598212d933ba",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "7617e5cea99145fe14a37c548f361427075f3d4b02d05fd4913602984928168d",
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.21",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "MacAddress": "02:42:ac:11:00:15",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "3c22c37a786ec7365527bdecb203ef48f76204b5e52a9d8c9cbaca0308fa6a91",
                    "EndpointID": "7617e5cea99145fe14a37c548f361427075f3d4b02d05fd4913602984928168d",
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.21",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:11:00:15",
                    "DriverOpts": null
                }
            }
        }
    }
]
```

**Comment by josegonzalez** [comment]
When you run a temporary container, you need to use the `--rm` flag on the command to ensure it gets removed when its no longer being interacted with. Example:

```
dokku --rm run $APP $COMMAND
```

This currently isn't done by default but I think maybe I should set that up...

**Comment by tylercal** [comment]
That makes sense, thanks for the tip!

For others searching, I cleaned up the old containers by using `docker container prune`

**Comment by josegonzalez** [comment]
I'm going to split this up into a `dokku run` and `dokku run:detached`.

- Both commands will remove the containers by default
- `run:detached` will additionally add the `-d` flag (run the container in detached mode)
- Only `run` will use a tty when available
- A new `--no-tty` flag will be added to `run` to optionally disable a tty

**Comment by josegonzalez** [comment]
Closed by #4701
